### PR TITLE
Extraction of crypto

### PR DESCRIPTION
**Pull request by InsanusMokrassar**

### CODE REVIEW

**Comment by soywiz** [comment]
Some work on https://github.com/korlibs/korge/issues/2040

File: korge-foundation/build.gradle.kts [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies {
     //add("commonTestApi", project(":korge-test"))
     commonTestApi(libs.kotlinx.coroutines.test)
     commonMainApi(project(":korlibs-time"))
+    commonMainApi(project(":korlibs-crypto"))
```
Comment (soywiz) [inline korge-foundation/build.gradle.kts:26]: Likely:

```suggestion
    commonMainApi(project(":korlibs-crypto"))
    commonMainApi(project(":korlibs-math"))
    commonMainApi(project(":korlibs-memory"))
```

**Comment by soywiz** [comment]
Looking at CI, math will likely require require depending on `korlibs-memory`.

**Comment by soywiz** [comment]
To reduce the scope of the PR, could you consider doing a PR first splitting just `korlibs-crypto`? Then in a separate PR -memory could be splitted, and then math.

**Comment by InsanusMokrassar** [comment]
> Looking at CI, math will likely require require depending on `korlibs-memory` (or require copying the `arraycopy` functions as internal)

Yes, looks like it

File: korge-foundation/build.gradle.kts (hunk 1 above)
Comment (InsanusMokrassar) [reply korge-foundation/build.gradle.kts:26]: Rewrote to extract crypto only

File: korlibs-crypto/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
+import korlibs.*
+import korlibs.korge.gradle.generate.*
+
+description = "Korge Foundation Libraries"
```
Comment (soywiz) [inline korlibs-crypto/build.gradle.kts:5]: Korlibs Cryptography Library

File: korlibs-crypto/src/korlibs/memory/InternalCryptoArrays.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package korlibs.memory
+
+/** Copies [size] elements of [src] starting at [srcPos] into [dst] at [dstPos]  */
+internal fun internalArrayCopy(src: ByteArray, srcPos: Int, dst: ByteArray, dstPos: Int, size: Int) {
```
Comment (soywiz) [inline korlibs-crypto/src/korlibs/memory/InternalCryptoArrays.kt:4]: Can you keep the `arraycopy` name? and move this file to the `korlibs.memory.internal` package?

**Review by soywiz — COMMENTED** [review]
Minor things. Can you have a look?

**Comment by InsanusMokrassar** [comment]
> Minor things. Can you have a look?

Sure, will fix it asap

File: korlibs-crypto/build.gradle.kts (hunk 2 above)
Comment (InsanusMokrassar) [reply korlibs-crypto/build.gradle.kts]: Fixed

File: korlibs-crypto/src/korlibs/memory/InternalCryptoArrays.kt (hunk 3 above)
Comment (InsanusMokrassar) [reply korlibs-crypto/src/korlibs/memory/InternalCryptoArrays.kt]: Fixed

**Comment by soywiz** [comment]
Awesome! Thank you very much

**Pull request merged**